src/streamlit_extras/sandbox/__init__.py

```diff
@@ -0,0 +1,151 @@
+from __future__ import annotations
+
+import inspect
+import textwrap
+from typing import Callable, List
+
+from streamlit.components.v1 import html
+
+from .. import extra
+
+
+@extra
+def sandbox(
+    code: str | Callable[[], None],
+    stlite_version: str | None = None,
+    requirements: List[str] | None = None,
+    height: int = 700,
+    scrolling: bool = False,
+) -> None:
+    """Execute untrusted Streamlit code in a sandboxed environment.
+
+    This function allows you to execute untrusted Streamlit code inside the user's web browser
+    by using stlite (https://github.com/whitphx/stlite) instead of the App server. This is useful
+    for apps that generate  and execute Streamlit (or Python) code at runtime based on some user
+    instructions. Doing this inside the main Streamlit app would be unsafe since the user could
+    execute arbitrary code on the server.
+
+    Parameters
+    ----------
+
+    code : str or callable
+        The code to execute. This can either be a string containing the code or a function.
```

Comment:
So great! I love being able to pass either a string or a function.

Reply:
One missing functionality here is to be able to transfer (serializable) objects from the server instance to the sanboxed instance. This would be pretty sweet and would allow to build a safe version of the LLM-powered chart builder app :)